pkgs/development/go-modules/generic/default.nix

```diff
@@ -198,7 +198,7 @@ go.stdenv.mkDerivation (
      ln -s "${dep.src}" "$d/src/${dep.goPackagePath}"
   ''
   ) goPath) + ''
-    export GOPATH="$d:$GOPATH"
+    export GOPATH=${lib.concatStringsSep ":" ( ["$d"] ++ ["$GOPATH"] ++ ["$PWD"] ++ extraSrcPaths)}
```

Comment:
Usually this is up to the user to set this and people might have different opinions how they organize their `GOPATH`.
Some may create one GOPATH per project others might use one global GOPATH for all go code (https://github.com/golang/go/wiki/GOPATH#use-a-single-gopath).